### PR TITLE
Change pyramid_level_inputs to have string keys

### LINKED ISSUE #1820: Change `pyramid_level_inputs` to have string keys
Currently `pyramid_level_inputs` uses `int` keys, but this breaks serialization (#1812). We patched this up in #1815 at the Task level, but a better fix is to change this attribute to use string keys in all the backbones ([example](https://github.com/keras-team/keras-cv/blob/ef7707f87362eb340190c71baa5b0b5d7f344a40/keras_cv/models/backbones/resnet_v2/resnet_v2_backbone.py#L359)) and in its docstring ([link](https://github.com/keras-team/keras-cv/blob/ef7707f87362eb340190c71baa5b0b5d7f344a40/keras_cv/models/backbones/backbone.py#L159)).

After that we can remove the ad hoc fix in the OD tasks themselves.

### PR DESCRIPTION
**Pull request by avinair**
This PR changes the data type of the keys in the pyramid_level_inputs dictionary from int to string since int keys do not seem to work nicely with TensorFlow model saving flow.

Fixes #1820 

## Who can review?
@jbischof @ianstenbit

### CODE REVIEW

File: keras_cv/models/legacy/object_detection/faster_rcnn/faster_rcnn.py
Comment (avinair) [inline keras_cv/models/legacy/object_detection/faster_rcnn/faster_rcnn.py:62]: [Leaving a comment here since Github isn't allowing me to make one at the actual lines but I'm referring to lines 78-84]
I could not change the output of this function to use the new string naming convention since it requires some changes to the following:
- Anchor Generator in lines 293 and 296
- RoI Aligner - https://github.com/keras-team/keras-cv/blob/9d62238f7b99f18eaa1d83727e4dd8d5dd5a920a/keras_cv/layers/object_detection/roi_align.py#L195-L204

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py
Comment (ianstenbit) [inline keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py:147]: let's use f-strings throughout for these (e.g `f"P{str(index+2)}"`)

File: keras_cv/models/legacy/object_detection/faster_rcnn/faster_rcnn.py
Comment (ianstenbit) [reply keras_cv/models/legacy/object_detection/faster_rcnn/faster_rcnn.py:62]: I think it would be best to update this + the ROIAligner so that we have consistency across the package.

If you'd like to also update the RetinaNet FPN, feel free to update that one as well (which currently outputs a tuple instead of a dict)

File: keras_cv/models/object_detection/retinanet/retinanet.py
Comment (avinair) [inline keras_cv/models/object_detection/retinanet/retinanet.py:195]: @ianstenbit Let me know if you want me to handle the output of the FeaturePyramid differently. I wasn't sure what would be the best way to handle changing the output to a dict and decided to go with this option because of how it is handled in the [yolo_v8_detector](https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py#L409). I believe dicts are ordered in Python3.6 and beyond, so this should be deterministic.

File: keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py
Comment (jbischof) [inline keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py:51]: This list should be used everywhere or the levels should be inlined everywhere.

File: keras_cv/models/object_detection/retinanet/retinanet.py
Comment (ianstenbit) [reply keras_cv/models/object_detection/retinanet/retinanet.py:195]: Yes, as of Python3.7 I think, dicts are guaranteed to be insertion-ordered.

That said, it may be nice to be a bit more overt where we can, e.g. by saying something like:
```
for pyramid_level in ["P1", "P2", ...]:
    feature = features[pyramid_level]
    ...
```

Comment (avinair) [reply keras_cv/models/object_detection/retinanet/retinanet.py]: Done

File: keras_cv/layers/object_detection/roi_align.py
Comment (ianstenbit) [inline keras_cv/layers/object_detection/roi_align.py:201]: Is this strictly required? If we take the max using the strings, shouldn't we get the same result due to the String sort order matching the int order in this case?

**Review by ianstenbit — APPROVED** [review]
Thanks! Looks good to me.

I'll wait for an approval from Jonathan too

File: keras_cv/layers/object_detection/roi_align.py
Comment (avinair) [reply keras_cv/layers/object_detection/roi_align.py:201]: You are right, if you use max using strings you would get the same result and that's what I initially did. However, it seemed easier to convert it to an int list upfront to avoid messing with the operations on Lines 219, 260, 263 & 280 (the line numbers are the numbers after the edits).

**Review by jbischof — APPROVED** [review]
Looks great, thank you!

**Pull request merged**
